### ISSUE TITLE
Create a WIP commit before pushing.

### ISSUE THREAD

**Issue by murjax** [MichaelDimmitt/git_push_daemon#2]
Sometimes we forget to make commits. If we're pushing, why not make a commit with all changes? These are easily reverted if the user does not want these changes, but at least they'll be recorded. My suggestion is to push these changes to a new temp_wip branch and push to Github. This way it's safe and won't interfere with the workflow of anything else.

**Comment by ckchessmaster** [comment]
I agree this is actually more often the case for me if I have to leave in a hurry for something.

**Comment by MichaelDimmitt** [comment]
I am going to open a new issue describing this because it will be an optional feature. 
What I am thinking: 
1. WIP merge branch ... (like an alternative dev branch)
2. a bunch of WIP-temp# branches. ex: WIP-temp1, WIP-temp2, WIP-temp3 

WIP merge branch will use function rebase master every single time. If a conflict happens it will perform git reset hard origin/WIP-merge. Since the WIP-merge is never not pushed it will always be up to date.

**Comment by MichaelDimmitt** [comment]
@murjax, in terms of the temp branches. 

I am thinking to use command:
`git checkout -b WIP-temp$count` this creates a new branch but fails if one already exists.
upon failure, increment the counter. 
until a success. 

When people merge these branches back in they might delete branches.
This method supports that delete action!

Though I will have to pay attention to this action for time consumption concerns.
